src/model/neuroevolution/individual.py

```diff
@@ -27,35 +27,82 @@ def get_random_layer_index(self) -> int:
     def get_random_weight_index(neural_network_layer: Layer) -> Tuple[int, ...]:
         return tuple(np.random.randint(neural_network_layer.weights.shape))
 
+    @staticmethod
+    def get_random_weight_indexes(
+        layer: Layer, indexes_count: int
+    ) -> Tuple[np.ndarray, ...]:
+        """
+        Calculates indexes_count random weight indexes for a single layer.
+
+        Returned tuple is ready to be used immediately for indexing the layer's weights.
+        """
+        weights = layer.weights
+        return tuple(
+            np.random.randint(0, weights.shape, size=(indexes_count, weights.ndim)).T
+        )
+
     @staticmethod
     def get_random_bias_index(neural_network_layer: Layer) -> int:
         return cast(int, np.random.randint(neural_network_layer.biases.shape[0]))
 
     @staticmethod
-    def random_weight_mutation(individual: "ChildIndividual") -> None:
+    def get_random_bias_indexes(
+        neural_network_layer: Layer, indexes_count: int
+    ) -> np.ndarray:
+        return np.random.randint(
+            neural_network_layer.biases.shape[0], size=indexes_count
+        )
```

Comment:
It would be good if these functions delegated to `utils.numpy_random_index` or the call sites just used that function.

Reply:
Done